Stp.TestingApi/Controllers/ExerciseController.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Stp.Data;
+using Stp.Data.Entities;
+using Stp.TestingApi.Contracts;
+using Microsoft.EntityFrameworkCore;
+
+namespace Stp.TestingApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ExerciseController : ControllerBase
+    {
+        private readonly TestingDbContext _db;
+        public ExerciseController(TestingDbContext db)
+        {
+            _db = db;
+        }
+
+        /* https://localhost:5001/api/Exercise/GetExercisesByCategory */
+        [HttpGet(nameof(GetExercisesByCategory))]
+        public IEnumerable<ExerciseDto> GetExercisesByCategory(long exerciseCategoryId)
+        {
+            var res = _db.ExerciseList
+                .ToList().Where(x => x.ExerciseCategoryId == exerciseCategoryId)
+                .Select(x => new ExerciseDto()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Description = x.Description,
+                    Points = x.Points,
+                    DurationMinutes = x.DurationMinutes, 
+                    Complexity = (ExerciseComplexity)x.Complexity
+                });       
```

Comment:
ToList function should be called the last.
Calling ToList before Select and Where clauses leads to generating SQL like "select * from Exercise" which is not effective.
EF converts Linq query to SQL expression before calling materialization function (like ToList, ToArray, FirstOrDefault etc)